lib/hanami/extensions/view/part.rb

```diff
@@ -4,20 +4,67 @@ module Hanami
   module Extensions
     module View
       # @api private
+      # @since 2.1.0
       module Part
         def self.included(part_class)
           super
 
           part_class.extend(Hanami::SliceConfigurable)
-          part_class.include(StandardHelpers)
           part_class.extend(ClassMethods)
         end
 
         module ClassMethods
           def configure_for_slice(slice)
-            extend SliceConfiguredHelpers.new(slice)
+            const_set :PartHelpers, Class.new(PartHelpers) { |klass|
+              klass.configure_for_slice(slice)
+            }
           end
         end
+
+        # Returns an object including the default Hanami helpers as well as the user-defined helpers
+        # for the part's slice.
+        #
+        # Use this when you need to access helpers inside your part classes.
+        #
+        # @return PartHelpers
+        #
+        # @api public
+        # @since 2.1.0
+        def helpers
+          @helpers ||= self.class.const_get(:PartHelpers).new(context: _context)
+        end
+      end
+
+      # Standalone helpers class including both {StandardHelpers} as well as the user-defined
+      # helpers for the slice.
+      #
+      # Used used where helpers should be addressed via an intermediary object (i.e. in parts),
+      # rather than mixed into a class directly.
+      #
+      # @api private
+      # @since 2.1.0
+      class PartHelpers
+        extend Hanami::SliceConfigurable
+
+        include StandardHelpers
+
+        def self.configure_for_slice(slice)
+          extend SliceConfiguredHelpers.new(slice)
+        end
+
+        # @api public
+        # @since 2.1.0
+        attr_reader :_context
```

Comment:
@timriley Why is this Public API?
As per Ruby conventions, a name starting with an underscore is a warning sign for the reader not to use that method because it's private ("for real").

I understand that we want to expose the context, but can't we just make `context` as Public API? So it's cleaner and it follows the convention described above.

Reply:
@jodosha Inside hanami-view (previously dry-view), access to the context object is indeed public API, but we use this underscored version because the context is always located in objects where the user is supplying their own values, such as scopes/templates and parts.

Here's how we explain it in the parts documentation: https://dry-rb.org/gems/dry-view/0.7/parts/#accessing-the-context

In both the scope (templates) _and_ parts we actually provide `#context` as a convenience alias. But as soon as there is (a) a local in the scope or (b) wrapped value in a part that uses the `context` name, we preference that instead of our internal context object. This is why any library- or framework-provided code _must_ use `_context` instead. The non-prefixed `context` alias is there only as a nicety for user-land code.

Does that help explain things for you?